File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.java
import java.util.*;
class Solution {
	static int[][] arr;
	static List<Integer> list;
	static int snail_maxNum;
	static int dir;
    public int[] solution(int n) {
        int[] answer = {};
        arr = new int[n][n];
        list = new ArrayList<>();
        snail_maxNum = (n*(n+1))/2;
        int row_u = 0, column_l = 0;
        int column_r = n-1;
        int row_d = n - 1;
        //아래 옆 대각선 방향
        dir = 0;
        int idx = 0;
        while(idx < snail_maxNum) {
        	// 아래 방향
        	for(int i = row_u; i <= row_d && idx < snail_maxNum; i++)
        		arr[i][column_l] = ++idx;
        	column_l++;
        	// 오른쪽 방향
        	for(int i = column_l; i <= column_r && idx < snail_maxNum; i++) {
        		if(arr[row_d][i] != 0)
        			break;
        		arr[row_d][i] = ++idx;
        	}
        	row_d--;
        	column_r--;
        	// 대각선 방향 주의하기!
        	for(int i = 0; i <= row_d && idx < snail_maxNum; i++) {
        		if(arr[row_d-i][column_r-i] != 0) 
        			break;
        		arr[row_d-i][column_r-i] = ++idx;
        	}
        	row_u += 2;
        	column_r --;
        	}
        for(int i = 0; i < arr.length; i++) {
        	for(int j = 0; j < arr.length; j++) {
        		if(arr[i][j] != 0)
        			list.add(arr[i][j]);
        	}
        }
        answer = new int[list.size()];
        for (int i = 0 ; i < list.size() ; i++) 
            answer[i] = list.get(i);
        
        return answer;
    }
}

File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.py
# 삼각 달팽이
def solution(n):
    # 2차원 배열 만드는 방법
    array = [[0] * n for i in range(n)]
    finish_num = int(n * (n+1) / 2)
    num = 1
    i = 0
    j = 0
    array[i][j] = num
    while num < finish_num:
        while i + 1 < n and array[i+1][j] == 0:
            i += 1
            num += 1
            array[i][j] = num
        while j + 1 < n and array[i][j+1] == 0:
            j += 1
            num += 1
            array[i][j] = num
        while (i-1 >= 0 and j-1 >= 0) and array[i-1][j-1] == 0:
            i -= 1
            j -= 1
            num += 1
            array[i][j] = num
    # 이렇게 하면 되는구나?
    answer = [num for row in array for num in row if num != 0]
    return answer